.github/workflows/create-update-release.yaml

```diff
@@ -0,0 +1,32 @@
+---
+name: Update Major Version of Prefect Deploy Action
+on:
+  push:
+    tags:
+      # Match the version format to not catch vx major releases
+      - 'v*.*.*'
+permissions: {}
+jobs:
+  update-create-major-version:
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Release or Update Major Version
+        run: |
+          export MAJOR_VERSION=$(echo ${{ github.ref_name }} | cut -d '.' -f 1)
+          echo "Releasing major version ${MAJOR_VERSION}"
+          if git show-ref --tags --verify --quiet "refs/tags/${MAJOR_VERSION}"; then
+            echo "Tag ${MAJOR_VERSION} exists, bumping to match with the latest release"
+            git tag ${MAJOR_VERSION} -f
+            git push origin ${MAJOR_VERSION} -f
+          else
+            echo "Tag ${MAJOR_VERSION} does not exist. Creating a new tag and release."
+            export RELEASE_SHA=$(git rev-parse HEAD)
+            gh release create "${MAJOR_VERSION}" --title "${MAJOR_VERSION}" --generate-notes --target "${RELEASE_SHA}" --latest
+          fi
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I think this can be ref'd by $GITHUB_TOKEN but i could be wrong

Reply:
https://github.com/elgohr/Github-Release-Action?tab=readme-ov-file#github-release-action I was going off of this referenced by the github actions docs and I _think_ it works since I tested a major bump in my own repo but if theres a better way I'm def down to change!